### ISSUE TITLE
Null SourceCollection error with update 0.6.2 and 0.7.0

### ISSUE THREAD

**Issue by Nils-1st-retail** [php-translation/symfony-bundle#224]
I have a working installation or 0.6.1 but after an update to 0.7.0 my installation is broken. I tried 0.6.2 but experience the same error as on 0.7.0

The error I get is:
`Type error: Argument 2 passed to Translation\Extractor\Visitor\Twig\Worker::work() must be an instance of Translation\Extractor\Model\SourceCollection, null given, called in [...]/vendor/php-translation/extractor/src/Visitor/Twig/TwigVisitor.php on line 63`

I'm not sure what exactly the problem is other than it seems the SourceCollection isn't initialised before being used.

**Comment by Nyholm** [comment]
That is strange. Thank you for the report. 
I do not remember any changes regarding to Twig.

Could you debug this and see why the `SourceCollection` is not initialized? 

Also, what version of Symfony are you using?

**Comment by Nils-1st-retail** [comment]
Hmm I can see that the init function on the BaseVisitor is never called. 
But I think I have something else wrong, because I get the error while visiting one of my webpages, I should never end up inside the extractor in that case?
Or am I missing something?

**Comment by pimolo** [comment]
Which command causes your error?
Could you dump there your configuration please?

**Comment by DidierLmn** [comment]
I'm having the same issue. 
I use [J-Mose/CommandSchedulerBundle](https://github.com/J-Mose/CommandSchedulerBundle), when accessing this bundles front via the 'command-scheduler/list' route, there are no problems. However when clicking the new button at their 'command-scheduler/detail/new' route, the php-translation/symfony-bundle throuws this very same error.

I can confirm this issue is present in 0.6.2 and 0.7.0 of this bundle, but not in 0.6.1.
I'm also running Symfony 3.4.8

**Comment by FabienPapet** [comment]
Got the same here: 
steps to reproduce : 
```sh
composer create-project symfony/website-skeleton my-project
cd my-project
composer require php-translation/symfony-bundle
composer require jmose/command-scheduler-bundle
php bin/console d:s:u --force


php bin/console server:run 
```

Then go to http://localhost:8000/command-scheduler/detail/new

**Comment by FabienPapet** [comment]
ping @pimolo

**Comment by FabienPapet** [comment]
Any updates regarding this issue ?

**Comment by DidierLmn** [comment]
I've done some investigating, and the issue may be from the [php-translation/extractor package](https://github.com/php-translation/extractor), when calling  [`Translation\Extractor\Visitor\Twig\TwigVisitor.php` on line 62](https://github.com/php-translation/extractor/blob/master/src/Visitor/Twig/TwigVisitor.php#L61). 

`$this->collection` is `null` there, as well as `$this->file`. 
These variables don't get initialized in the `TwigVisitor` constructor, but get initialized in the parent class, [`BaseVisitor` in the `init()` function](https://github.com/php-translation/extractor/blob/master/src/Visitor/BaseVisitor.php#L45).
However this function appears to not be called anywhere, hence `$collection` and `$file` are `null`.

**Comment by FabienPapet** [comment]
This was introduced in https://github.com/php-translation/extractor/commit/a423a523b942ba97892f944a6ae53e9561f4ea30#diff-aa3ebddd7f502b7edf90902e1af128ac

**Comment by Nyholm** [comment]
Hm. Okey. Thank you Fabien. I’ll do some investigation this week.

**Comment by Lo-X** [comment]
Hello, any update on that issue? I'm having the same error but with Symfony 2.8.18

**Comment by Lo-X** [comment]
Hello again, I managed to get rid of this error in command lines by disabling the symfony_profiler in dev configuration. This is not optimal but maybe it will help locating the issue.

```
translation:
    symfony_profiler:
        enabled: false
    webui:
        enabled: false
```

**Comment by Nyholm** [comment]
Thank you for all the investigation and research. I've finally manage to find the issue (based on the comments here). 

See PR: https://github.com/php-translation/extractor/pull/115

**Comment by Nyholm** [comment]
This is fixed if you update to 1.6.0 of php-translation/extractor. 

https://github.com/php-translation/extractor/releases/tag/1.6.0
